src/raftstore/coprocessor/region_snapshot.rs

```diff
@@ -49,50 +49,55 @@ impl RegionSnapshot {
         &self.region
     }
 
-    pub fn iter(&self) -> RegionIterator {
-        RegionIterator::new(self.snap.new_iterator(), self.region.clone())
+    pub fn iter(&self, upper_bound: Option<&[u8]>) -> RegionIterator {
+        if upper_bound.is_none() {
+            RegionIterator::new(self.snap.new_iterator(Some(keys::enc_end_key(self.get_region())
+                                    .as_slice())),
+                                self.region.clone())
+        } else {
+            RegionIterator::new(self.snap.new_iterator(upper_bound), self.region.clone())
```

Comment:
It's better to add a new function in RegionIterator to handle the encode/decode thing.